fili-core/src/main/java/com/yahoo/bard/webservice/data/filterbuilders/ConjunctionDruidFilterBuilder.java

```diff
@@ -88,6 +96,80 @@ public Filter buildFilters(Map<Dimension, Set<ApiFilter>> filterMap) throws Dime
     protected abstract Filter buildDimensionFilter(Dimension dimension, Set<ApiFilter> filters)
             throws DimensionRowNotFoundException;
 
+    /**
+     * Splits a set of {@link ApiFilter}s into two groups of positive and negative filters.
+     * <p>
+     * An {@link ApiFilter} is defined to be positive if its filter operation is one of
+     * <ul>
+     *     <li> {@link FilterOperation#in}
+     *     <li> {@link FilterOperation#startswith}
+     *     <li> {@link FilterOperation#contains}
+     *     <li> {@link FilterOperation#eq}
+     * </ul>
+     * An {@link ApiFilter} is defined to be negative if its filter operation is {@link FilterOperation#notin}.
+     *
+     * @param filters  A set of API filters that are to be splitted
+     *
+     * @return a pair whose left is positive filter and whose right is negative filers
+     */
+    protected static Pair<Set<ApiFilter>, Set<ApiFilter>> splitApiFilters(Set<ApiFilter> filters) {
+        Set<ApiFilter> positiveFilters = new HashSet<>();
+        Set<ApiFilter> negativeFilters = new HashSet<>();
+
+        for (ApiFilter filter : filters) {
+            if (FilterOperation.notin.equals(filter.getOperation())) {
```

Comment:
(style) If you wanted to do this in a streamy way, you could use a grouping collector I believe.